src/tyr/isochrone_serializer.cc

```diff
@@ -142,5 +284,68 @@ std::string serializeIsochrones(const Api& request,
 
   return ss.str();
 }
+std::string serializeIsochronePbf(Api& request,
+                                  std::vector<contour_interval_t>& intervals,
+                                  const contours_t& contours) {
+  // construct pbf output
+  Isochrone& isochrone = *request.mutable_isochrone();
+
+  // construct contours
+  for (size_t isoline_index = 0; isoline_index < contours.size(); ++isoline_index) {
+    const auto& contour = contours[isoline_index];
+    const auto& interval = intervals[isoline_index];
+
+    auto* interval_pbf = isochrone.mutable_intervals()->Add();
+    interval_pbf->set_metric(std::get<2>(interval) == "time" ? Isochrone::time : Isochrone::distance);
+
+    interval_pbf->set_metric_value(std::get<1>(interval));
+
+    // for each feature
+    for (const auto& feature : contour) {
+      grouped_contours_t groups = GroupContours(true, feature);
+
+      // for each group of rings (first is outer, rest is inner)
+      for (std::vector<const contour_t*> group_ptr : groups) {
+        auto* contour_pbf = interval_pbf->mutable_contours()->Add();
+
+        // construct a geometry
+        for (const std::list<PointLL>* ring : group_ptr) {
+          std::cerr << "Rings: " << ring->size() << std::endl;
+
+          auto* geom = contour_pbf->mutable_geometries()->Add();
+          for (PointLL pair : *ring) {
+            geom->add_coords(round(pair.lng() * 1e6));
+            geom->add_coords(round(pair.lat() * 1e6));
+          }
+        }
+      }
+    }
+  }
+
+  return serializePbf(request);
+}
+
+} // namespace
+
+namespace valhalla {
+namespace tyr {
+
+std::string serializeIsochrones(Api& request,
+                                std::vector<midgard::GriddedData<2>::contour_interval_t>& intervals,
+                                midgard::GriddedData<2>::contours_t& contours,
+                                bool polygons,
+                                bool show_locations) {
+
+  switch (request.options().format()) {
+    case Options_Format_pbf:
+      return serializeIsochronePbf(request, intervals, contours);
+    case Options_Format_json:
+      if (polygons)
```

Comment:
This still needs to change, probably should keep the current serialization as the default and only use the new method when explicitly requested by the user.